### PR TITLE
fix: save ak to txt on signup for android without pdf reader [branch …

### PR DESCRIPTION
**Pull request by Karim94**
…ch13122]

#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/13122/android-improved-handling-of-ak-pdf-in-account-creation

#### Testing instructions  
There were no designs for this, but they're unnecessary. Important thing is to make sure that the flow for saving AK on an Android device without a pdf reader is the same as saving it on one that has a reader.
Flow:
1- In sign up account key backup screen, press save PDF
2- Create, save and open a txt file named `*username*-*appName*}.txt`
3- File contains labelled Username and Account Key
4- User presses back to exit the txt file and continues account creation flow

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: app/components/signup/signup-state.js [hunk 1]
```diff
@@ -111,8 +111,9 @@ class SignupState extends RoutedState {
         });
     }
 
-    get backupFileName() {
-        return `${this.username}-${tx('title_appName')}.pdf`;
+    @action.bound
```
Comment (seavan) [inline app/components/signup/signup-state.js]: use this.backupFileName instead, no action.bound

File: app/components/signup/signup-state.js [hunk 2]
```diff
@@ -140,13 +141,11 @@ class SignupState extends RoutedState {
...
-                const path = `${directory}/${username}-Peerio-Account-Key.txt`;
-                const content = `Peerio Username: ${username}\nPeerio Account Key: ${passphrase}`;
+                fileSavePath = config.FileStream.getTempCachePath(backupFileName('txt'));
+                const content = `${tx('title_appName')} Username: ${username}\n${tx('title_appName')} Account Key: ${passphrase}`;
 
-                RNFS.writeFile(path, content, 'utf8')
-                    .then(() => FileOpener.open(path, 'text/*', path))
-                    .catch((err) => console.log(err.message));
+                await RNFS.writeFile(fileSavePath, content, 'utf8');
+                await FileOpener.open(fileSavePath, 'text/*', fileSavePath);
```
Comment (seavan) [inline app/components/signup/signup-state.js]: rework the flow so that we reach keyBackedUp from one path.

Comment (Karim94) [reply app/components/signup/signup-state.js]: fixed

File: app/components/signup/signup-state.js (hunk 1 above)
Comment (Karim94) [reply app/components/signup/signup-state.js]: fixed

**Comment by ebilg** [comment]
can be merged

**Pull request merged**